miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/TissueOriginController.java

```diff
@@ -98,32 +96,27 @@ private static TissueOriginDto writeUrls(TissueOriginDto tissueOriginDto, UriCom
   }
 
   @RequestMapping(value = "/tissueorigin", method = RequestMethod.POST, headers = { "Content-type=application/json" })
-  @ResponseBody
-  public ResponseEntity<?> createTissueOrigin(@RequestBody TissueOriginDto tissueOriginDto, UriComponentsBuilder b,
+  @ResponseStatus(HttpStatus.CREATED)
+  public @ResponseBody TissueOriginDto createTissueOrigin(@RequestBody TissueOriginDto tissueOriginDto, UriComponentsBuilder b,
       HttpServletResponse response) throws IOException {
     TissueOrigin tissueOrigin = Dtos.to(tissueOriginDto);
     Long id = tissueOriginService.create(tissueOrigin);
-    UriComponents uriComponents = b.path("/tissueorigin/{id}").buildAndExpand(id);
-    HttpHeaders headers = new HttpHeaders();
-    headers.setLocation(uriComponents.toUri());
-    return new ResponseEntity<>(headers, HttpStatus.CREATED);
+    return Dtos.asDto(tissueOriginService.get(id));
   }
 
   @RequestMapping(value = "/tissueorigin/{id}", method = RequestMethod.PUT, headers = { "Content-type=application/json" })
-  @ResponseBody
-  public ResponseEntity<?> updateTissueOrigin(@PathVariable("id") Long id, @RequestBody TissueOriginDto tissueOriginDto,
+  public @ResponseBody TissueOriginDto updateTissueOrigin(@PathVariable("id") Long id, @RequestBody TissueOriginDto tissueOriginDto,
       HttpServletResponse response) throws IOException {
     TissueOrigin tissueOrigin = Dtos.to(tissueOriginDto);
     tissueOrigin.setId(id);
     tissueOriginService.update(tissueOrigin);
-    return new ResponseEntity<>(HttpStatus.OK);
+    return Dtos.asDto(tissueOriginService.get(id));
   }
 
   @RequestMapping(value = "/tissueorigin/{id}", method = RequestMethod.DELETE)
-  @ResponseBody
-  public ResponseEntity<?> deleteTissueOrigin(@PathVariable("id") Long id, HttpServletResponse response) throws IOException {
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  public void deleteTissueOrigin(@PathVariable("id") Long id, HttpServletResponse response) throws IOException {
     tissueOriginService.delete(id);
```

Comment:
Could it return a friendlier error message if the tissue origin is in use? Current message is 
```
could not execute statement; SQL [n/a]; constraint [null]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement
```